### PR TITLE
Sensor Support

### PR DESCRIPTION
**Pull request by JBassett**
This is super rough sensor support.  Battery level and charging "working".
https://developers.home-assistant.io/docs/en/app_integration_sensors.html

fixes: https://github.com/home-assistant/home-assistant-android/issues/122
fixes: https://github.com/home-assistant/home-assistant-android/issues/128

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/launch/LaunchActivity.kt [hunk 1]
```diff
@@ -62,4 +68,12 @@ class LaunchActivity : AppCompatActivity(), LaunchView {
     override fun getMenuInflater(): MenuInflater {
         return LokaliseMenuInflater(this)
     }
+
+    private fun startSensorWorker() {
+        val sensorWorker =
+            PeriodicWorkRequestBuilder<SensorWorker>(15, TimeUnit.MINUTES)
```
Comment (balloob) [inline app/src/main/java/io/homeassistant/companion/android/launch/LaunchActivity.kt]: Is this something we want to do periodically or only when:
 - we send a location event
 - we haven't send a location event in 30 min
 - we change charging type (non-charging -> charging)

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/launch/LaunchActivity.kt]: So in this initial implementation I was going for just every 15 ish minutes update all sensors.  I plan on listening for change events wherever possible to trigger an individual sensor to update.  I didn't want to link this functionality to the background location updates initially but there is nothing saying we can't.

**Comment by dshokouhi** [comment]
With this change will we need to redo the integration or restart HA to get the sensors to show up or will they register and appear after enabling?  Also if you could share a screenshot of the new settings that would be great :)

**Comment by JBassett** [comment]
Screenshot_20200208-155536

Screenshot_20200208-154334

**Comment by JBassett** [comment]
image

**Comment by JBassett** [comment]
Once you enable the sensors in settings they will be registered and just appear.  No restarts or resets needed.

**Comment by balloob** [comment]
Not sure if I'm a fan of making the sensors configurable. How will we remove them from Home Assistant if we disable them?

About the timing, could we do:

 - Send sensor data to Home Assistant when the location updates (and we already woke up the device)
 - Set a timer to send a location update 15 minutes after we send the last one. If a zone change triggers a location update before, we reset the timer.

**Comment by dtorner** [comment]
Talking just about the battery sensor, I would prefer to get it updated when it changes to a "charging" or "not_charging" state as well as the other timings, that way users could use the sensors for automations. Right now I use a custom app to send the battery state back to hass, when I wake up and disconnect the device from charging I turn on a light and get a weather telegram message...

**Comment by dshokouhi** [comment]
I think we should definitely trigger an update based on the charging status and when the wifi connection status changes (i.e. connecting/disconnecting/switching AP's).

**Comment by balloob** [comment]
Please do not comment with feature requests on PRs. Also [read all the comments](https://github.com/home-assistant/home-assistant-android/pull/336#discussion_r374112717) 🙄

**Comment by JBassett** [comment]
Removed settings around the sensors and added logic stated above.

**Pull request merged**